### PR TITLE
Java issues - OSAC-42, OSAC-43, OSAC-45

### PR DESCRIPTION
**Pull request by uipathact**
Fixed the following issues:

- problem invoking generic functions
- problem with null parameters
- problem with empty arrays

### CODE REVIEW

File: Activities/Java/Invoke/src/com/uipath/dotnetjavatypes/ArrayTypeSerializer.java [hunk 1]
```diff
@@ -19,16 +19,25 @@
...
-            return new JavaObject(new EmptyClass(), boolean.class);
+            try {
+                runtimeArrayType = obj.getString("runtime_arrayType");
+            }
+            catch (JSONException e) {  }
+            return EmptyTypeSerializer.DeserializeToNullJavaObject(runtimeArrayType);
         }
         Class<?> arrayType = getArrayType(context, array);
         int length = array.length();
+        if (array.length() == 0){
```
Comment (adrian-zanescu) [inline Activities/Java/Invoke/src/com/uipath/dotnetjavatypes/ArrayTypeSerializer.java:38]: use length from above :)

File: Activities/Java/UiPath.Java.Activities/JavaActivityWithParameters.cs [hunk 2]
```diff
@@ -48,6 +48,17 @@ protected List<object> GetParameters(AsyncCodeActivityContext context)
             List<object> parameters = ParametersList.Get(context) ?? Parameters.Select(arg => arg.Get(context)).ToList();
             return parameters;
         }
-
+        protected List<Type> GetParameterTypes(AsyncCodeActivityContext context, List<object> parameters)
+        {
+            List<Type> parameterTypes = new List<Type>();
+            for(int index=0;index<Parameters.Count;index++)
+            {
+                if(null != parameters.ElementAt(index))
```
Comment (adrian-zanescu) [inline Activities/Java/UiPath.Java.Activities/JavaActivityWithParameters.cs:56]: parameter[index]. also spaces inside the for expression

File: Activities/Java/Invoke/src/com/uipath/dotnetjavatypes/ArrayTypeSerializer.java (hunk 1 above)
Comment (uipathact) [reply Activities/Java/Invoke/src/com/uipath/dotnetjavatypes/ArrayTypeSerializer.java]: Fixed

File: Activities/Java/UiPath.Java.Activities/JavaActivityWithParameters.cs (hunk 2 above)
Comment (uipathact) [reply Activities/Java/UiPath.Java.Activities/JavaActivityWithParameters.cs]: fixed

**Review by adrian-zanescu — APPROVED** [review]

**Pull request merged**